src/sentry/identity/slack/provider.py

```diff
@@ -12,25 +12,46 @@ class SlackIdentityProvider(OAuth2Provider):
 
     # This identity provider is used for authorizing the Slack application
     # through their Bot token (or legacy Workspace Token if enabled) flow.
-    oauth_authorize_url = "https://slack.com/oauth/authorize"
 
     oauth_scopes = ("identity.basic", "identity.email")
 
+    def get_oauth_authorize_url(self):
+        if self.use_wst_app:
+            return "https://slack.com/oauth/authorize"
+        return "https://slack.com/oauth/v2/authorize"
+
+    @property
+    def use_slack_v2(self):
+        # need to import here because of dependency issues
+        # would be nice to make the pipeline class have a use_slack_v2
+        # method but we use the same pipeline class everywhere
+        from sentry.integrations.slack.utils import use_slack_v2
+
+        return use_slack_v2(self.pipeline)
+
+    @property
+    def use_wst_app(self):
+        return settings.SLACK_INTEGRATION_USE_WST and not self.use_slack_v2
+
     # XXX(epurkhiser): While workspace tokens _do_ support the oauth.access
     # endpoint, it will no include the authorizing_user, so we continue to use
     # the deprecated oauth.token endpoint until we are able to migrate to a bot
     # app which uses oauth.access.
     def get_oauth_access_token_url(self):
         return (
             "https://slack.com/api/oauth.token"
-            if settings.SLACK_INTEGRATION_USE_WST
-            else "https://slack.com/api/oauth.access"
+            if self.use_wst_app
+            else "https://slack.com/api/oauth.v2.access"
         )
```

Comment:
Want to make this consistent with the other gets, instead of using the ternary style use

```
if x:
   return ...
return ...
```